cms/djangoapps/api/docs/decisions/001-course-apps.rst

```diff
@@ -0,0 +1,222 @@
+            # to see if this app can be enabled/configured for this course.
+            return True
+
+        @classmethod
+        def is_enabled(cls, course_key):
+            # Some logic to check if the app is enabled for this course
+            # This will not vary from user-to-user in studio.
+            return True
+
+        @classmethod
+        def set_enabled(cls, course_key, user, enabled):
+            # Some logic to enable the app for this course.
+            # The user here isn't passed on for permission checking, but just
+            # for logging/auditing.
+            return enabled
+
+        @classmethod
+        def get_allowed_operations(cls, course_key, user):
+            # This should return a dictionary with at least the `enable` and `configure` keys.
+            return {
+                'enable': can_user_enable(course_key, user),
+                'configure': can_user_configure(course_key, user),
+            }
+
+
+For existing features that need to be exposed as Course Apps, we can create
+such a class and have these class methods call back to the existing code for
+availability checks and enabled checks.
+
+Course Apps API
+---------------
+
+Each app has some associated metadata:
+
+- **id**: (string) A unique identifier for the app.
+- **name**: (string) A friendly name for the app that can be shown in the UI.
+- **description**: (string) A friendly description of what the app does, to be shown to
+  users in the UI.
+- **enabled**: (boolean) Is this app enabled for the current course.
+- **allowed_operations**: (dictionary) Apps can potentially enable/disable certain
+  operations. The following operations should be specified for all apps:
+
+    - **enable**: (boolean) Can the current user enable/disable this app.
+    - **configure**: (boolean) Can the current user can configure this app.
+
+  If an app doesn't have any configuration it can set the ``configure`` to false
+  and the UI will simply not show a configuration option for that app.
+- **legacy_link**: (string) If available, this will point to the legacy link for
+  configuring the app. This can be provided as a fallback while the new UX is
+  still in development.
+
+The is also the structure that will form the basis of the API's response:
+
+.. code-block:: python
+
+    {
+        'id': 'courseapp',
+        'enabled': False,
+        'name': 'Course App',
+        'description': 'A sample Course App for use as documentation.',
+        'allowed_operations': {
+            'enable': True,
+            'configure': True,
+            'edit_lti_config': False,
+        },
+        'legacy_link': 'https://studio.example.com/course_id/app-page',
+    }
+
+
+This API can be hosted at: ``/course_apps/v1/apps/{course_id}/``
+
+A ``GET`` request to this API will return an array of objects with the above
+structure. A ``PATCH`` request to the same endpoint with just the ``id`` and the
```

Comment:
Let's break down the `GET` and `PATCH` calls here. 
```
GET ``/course_apps/v1/apps/{course_id}/``

A ``GET`` request to this API will return an array of objects with the above structure.

PATCH ``/course_apps/v1/apps/{course_id}/``

 A ``PATCH`` request to the same endpoint with just the ``id`` and the 
`enabled` attribute can be used to enable/disable the app if it's possible to do so.
```

Please explain what form the `id` will be passed? Request data? query string?  
Is it an app id? let's be explicit and call `id` => `app-id`.